### ISSUE TITLE
Grant Permissions during runtime in Deno programaticaly

### ISSUE THREAD

**Issue by platinumindustries** [denoland/deno#10797]
I'm creating a server in Deno that will be able to run user submitted scripts as "applications".
 
The script/application will be imported dynamically (first time running the script) and cached by deno for subsequent requests.  

The script/application should by default have `read` access to its folder and only that specific folder. (It needs this permission so that it can run in the first place since its dynamically imported).

To make sure that the foreign scripts/applications don't violate the read permission, (bear in mind that permissions granted apply to the whole server, not just the module/script that requested it), I have only one foreign script/application running at any given time. Once it's done, i revoke its permissions in-case i need to call another foreign script/application during the same request.

Seeing that for each script that I ran I have to grant and deny permissions per request, and I know the permission each script needs ahead of time, is there a way to programaticaly Grant a permission? 

Here is a portion of my entry point to load the scripts/apps:

 ```
this.router.get("/:application", composeMiddleware([err, setPermissions]), async (context, next) => { 

            let application = await import(`./Applications/${context.params.application}/${context.params.application}.ts`)

            new application.default(context, next, this.router)  
    })
```

Here is my script to set permissions on each request per script:

```
async setPermissions(context: Record<string, any>, next: Function):Promise<void> {
        const desc = { name: "read", path: "./Applications/Preferences/Preferences.ts" } as const
        await Deno.permissions.request(desc)
        await next()
}
```

A way to grant the above permission without having to go to the terminal for every request after its reset would be welcome.

Note:  I have to revoke the permissions after each script is done just to  enforce a type of module level permission

**Comment by kitsonk** [comment]
If there was a way to programmatically grant permissions, that would defeat the whole security model, wouldn't it?  Any script could grant and revoke permissions as it sees fit.

**Comment by caspervonb** [comment]
Since you're describing these as scripts, spawning workers with permission sets seems more appropriate and you can do that today.

**Comment by trgwii** [comment]
@platinumindustries you DEFINITELY want process-isolation for user-submitted scripts, so they should run inside workers / separate deno processes anyway.

**Comment by lucacasonato** [comment]
Agree with @kitsonk. This proposal would allow the sanboxed code to  unsandbox itself by means of just requesting more permissions. Doesn't seem very sandboxed to me.

**Comment by platinumindustries** [comment]
@caspervonb @kitsonk  They might be several scripts and there is a possibility that they will run in parallel if i use workers/sub-processes. This in turn may add more overhead? during the requests. Is there a way to then give permissions on a per module basis?

**Comment by nayeemrmn** [comment]
> Is there a way to then give permissions on a per module basis?

No. A native binding call cannot reliably be associated with some lexical call site in JS, and such a system wouldn't be as intuitive as you'd think in a lot of cases. So all coded loaded into the JS sandbox is trusted uniformly.

**Comment by kitsonk** [comment]
And to add to what @nayeemrmn said, it is effectively a duplicate of #1639 then too.
